pkg/image/api/v1/trigger/trigger.go

```diff
@@ -0,0 +1,24 @@
+package trigger
+
+// ObjectFieldTrigger links a field on the current object to another object for mutation.
+type ObjectFieldTrigger struct {
```

Comment:
We don't validate triggers on deployment controllers.  I'm going to treat this behavior as undefined for now, and will document it as such.